lib/include/up-client-zenoh-cpp/client/upZenohClient.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _UP_ZENOH_CLIENT_H_
+#define _UP_ZENOH_CLIENT_H_
+
+#include <up-client-zenoh-cpp/transport/zenohUTransport.h>
+#include <up-client-zenoh-cpp/rpc/zenohRpcClient.h>
+
+class upZenohClient : public ZenohUTransport, public ZenohRpcClient {
+    public:
+        upZenohClient() {}
```

Comment:
I expected that to be true too, but the use of shared_ptr requires a public constructor.  Otherwise it won't compile.  I think that's going to have to be a trade off in this case, no real way to force the use of Instance.

Reply:
That is not good 
It means that someone can create an instance of upZenohClient and start calling the API without doing proper init , right?

Reply:
Yes, that is what it means.  I've been looking at alternative implementations, because I'm not happy with it either.

Reply:
Init is done in the constructor, but yes someone could construct this as a non-singleton.

One solution is to have the constructor take a parameter that is a type private to the class (i.e. put `struct ConstructToken{};` in the private section of upZenohClient) then have `upZenohClient::instance()` call `make_shared<…>(…, ConstructToken())`. That effectively makes the constructor private because the only way to provide one of the parameters is to have access to a private type.

Reply:
Pushed changes to use a private construct token.  Have a look and see what you guys think now.